atomic_reactor/plugins/build_orchestrate_build.py

```diff
@@ -231,40 +243,44 @@ def get_cluster_info(self, cluster, platform):
         osbs = OSBS(conf, conf)
         try:
             current_builds = self.get_current_builds(osbs)
-        except OsbsException as e:
+        except OsbsException:
             # If the build is canceled reraise the error
-            if isinstance(e.cause, BuildCanceledException):
-                raise e
-
-            self.log.exception("Error occurred while listing builds on %s",
-                               cluster.name)
-            return ClusterInfo(cluster, platform, osbs, self.UNREACHABLE_CLUSTER_LOAD)
+            raise
 
         load = current_builds / cluster.max_concurrent_builds
         self.log.debug('enabled cluster %s for platform %s has load %s and active builds %s/%s',
                        cluster.name, platform, load, current_builds, cluster.max_concurrent_builds)
         return ClusterInfo(cluster, platform, osbs, load)
 
-    def choose_cluster(self, platform):
-        config = get_config(self.workflow)
-        clusters = [self.get_cluster_info(cluster, platform) for cluster in
-                    config.get_enabled_clusters_for_platform(platform)]
-
-        if not clusters:
-            raise RuntimeError('No clusters found for platform {}!'
-                               .format(platform))
+    def get_clusters(self, all_clusters, cluster_fails, platform, retry_at):
+        ''' return clusters sorted by load '''
+
+        clusters = []
+        while all_clusters and not clusters:
+            for cluster in all_clusters:
+                if cluster_fails[cluster.name] >= self.max_cluster_fails:
+                    all_clusters.remove(cluster)
+                    continue
+                if cluster.name in retry_at:
+                    if datetime.datetime.now() < retry_at[cluster.name]:
+                        continue
+                try:
+                    clusters.append(self.get_cluster_info(cluster, platform))
+                except BuildCanceledException:
+                    raise
+                except OsbsException:
+                    cluster_fails[cluster.name] += 1
+                    retry_at[cluster.name] = datetime.datetime.now()\
+                        + timedelta(seconds=self.find_cluster_retry_delay)
+            if not clusters:
+                time.sleep((max(timedelta(seconds=0),
+                           min(retry_at.values()) - datetime.datetime.now())).seconds)
```

Comment:
The indentation alignment seems wrong here.

Reply:
How would you do it? This passes flake8

Reply:
What makes it hard to read is that the indented line does not line up with the parenthesis we're still in.

I'd personally prefer to the min and max calls in different statements, e.g.
```
next_retry_in = min(retry_at.values()) - datetime.datetime.now()
time.sleep(max(0, next_retry_in.seconds)
```